client/src/index.jsx

```diff
@@ -14,13 +15,34 @@ const Wrapper = Styled.div`
 class App extends React.Component {
   constructor() {
     super();
+    
+    this.state = {
+      locations: [],
+      materials: []
+    }
+    this.getProductById = this.getProductById.bind(this);
   }
   
+  getProductById() {
```

Comment:
I like that you made a function for data gathering, according to Leslie it may be better to call it in the constructor rather than componentDidMount because it will get started sooner

Reply:
Good point! I will do that